llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -4546,14 +4548,18 @@ ParseStatus AArch64AsmParser::tryParseZTOperand(OperandVector &Operands) {
     if (getParser().parseExpression(ImmVal))
       return ParseStatus::NoMatch;
     const MCConstantExpr *MCE = dyn_cast<MCConstantExpr>(ImmVal);
+    Operands.push_back(AArch64Operand::CreateImm(
```

Comment:
Can you change the order between the tests and the Operand push
```
   if (!MCE)
      return TokError("immediate value expected for vector index");
    Operands.push_back(AArch64Operand::CreateImm( ...);
```
It makes more sense because we are using MCE->getValue() when getting the value, if MCE does not exists it will fail